src/test/java/seedu/address/logic/commands/BuyCommandTest.java

```diff
@@ -0,0 +1,182 @@
+        assertFalse(buyCommand.equals(null));
+
+        // different types -> returns false
+        assertFalse(buyCommand.equals(new ClearCommand()));
+
+        // different index -> returns false
+        assertFalse(buyCommand.equals(new BuyCommand(INDEX_SECOND_COMPANY, BUY_ORANGE)));
+
+        // different Transaction -> returns false
+        assertFalse(buyCommand.equals(new BuyCommand(INDEX_FIRST_COMPANY, BUY_TOYS)));
+    }
+
+    /**
+     * A default model stub that have all the methods failing.
+     */
+    private class ModelStub implements Model {
+        private Company company;
+        private final UniqueCompanyList coys = new UniqueCompanyList();
+
+        public ModelStub(Company company) {
+            this.company = company;
+            coys.add(company);
+        }
+
+        @Override
+        public void setUserPrefs(ReadOnlyUserPrefs userPrefs) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public ReadOnlyUserPrefs getUserPrefs() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public GuiSettings getGuiSettings() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void setGuiSettings(GuiSettings guiSettings) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public Path getJeeqTrackerFilePath() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void setJeeqTrackerFilePath(Path jeeqTrackerFilePath) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void addCompany(Company company) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void setJeeqTracker(ReadOnlyJeeqTracker newData) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public ReadOnlyJeeqTracker getJeeqTracker() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public boolean hasCompany(Company company) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void deleteCompany(Company target) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void setCompany(Company target, Company editedCompany) {
+
+        }
+
+        @Override
+        public ObservableList<Company> getFilteredCompanyList() {
+            return coys.asUnmodifiableObservableList();
+        }
+
+        @Override
+        public void updateFilteredCompanyList(Predicate<Company> predicate) {
+
+        }
```

Comment:
I am not sure if it would be better if we added a `return;` statement within the function to make it apparent that this function is implemented and is meant to do nothing

Reply:
Instead of using a return statement, I think I would write a comment in the method body to state that it is being left empty intentionally. Same for the below comments. Let me know what you think.

Reply:
Okay, I think that works too